popup_offer/popup_offer.module

```diff
@@ -1,38 +0,0 @@
-<?php
-
-/**
- * Implements hook_preprocess_html().
- */
-function popup_offer_preprocess_html(&$variables)
-{
-  /* @var \Drupal\user\UserDataInterface $userData */
-  $userData = \Drupal::service('user.data');
-  $user = \Drupal::currentUser();
-  $product_builder = \Drupal::service('commerce_product.variation_field_renderer');
-  $builder = \Drupal::service('poppup_offer.special_discount');
-  $p_datas = $userData->get('popup_offer', $user->id(), 'discount_products');
```

Comment:
I don't think it's a good idea to use user->data for that.
what about anonymous?

Reply:
for anonymous is used session key, which is also unique. What is better way ?